test/infrastructure/inmemory/pkg/server/proxy/dial.go

```diff
@@ -33,85 +33,80 @@ import (
 
 const defaultTimeout = 10 * time.Second
 
-// Dialer creates connections using Kubernetes API Server port-forwarding.
+// Dialer opens connections via Kubernetes API server port-forward.
+//
+// Safe for concurrent use: each dial builds its own SPDY transport and http.Client
```

Comment:
Also if we think these comments are way better compared to the old ones, we probably should have the same at controlplane/kubeadm/internal/proxy/dial.go for consistency.